Sources/SwiftProtobuf/AnyMessageStorage.swift

```diff
@@ -16,276 +16,298 @@
+                        target = message
+                  }
+                  else {
+                        // Different type, serialize and parse.
+                        let data = try msg.serializedData(partial: true)
+                        target = try M(
+                              serializedData: data, extensions: extensions,
+                              partial: true)
+                  }
+
+            case .contentJSON(let contentJSON, let options):
+                  target = try unpack(
+                        contentJSON: contentJSON, options: options, as: M.self)
+                        as! M
+            }
       }
-      if !valueData.isEmpty {
-        try! visitor.visitSingularBytesField(value: valueData, fieldNumber: 2)
+
+      // Called before the message is traversed to do any error preflights.
+      // Since traverse() will use _value, this is our chance to throw
+      // when _value can't.
+      func preTraverse() throws {
+            switch state {
+            case .binary:
+                  // Nothing to be checked.
+                  break
+
+            case .message:
+                  // When set from a developer provided message, partial support
+                  // is done. Any message that comes in from another format isn't
+                  // checked, and transcoding the isInitialized requirement is
+                  // never inserted.
+                  break
+
+            case .contentJSON:
+                  // contentJSON requires a good URL and our ability to look up
+                  // the message type to transcode.
+                  if Google_Protobuf_Any.messageType(forTypeURL: _typeURL)
+                        == nil
+                  {
+                        // Isn't registered, we can't transform it for binary.
+                        throw BinaryEncodingError.anyTranscodeFailure
+                  }
+            }
       }
+}
 
-    case .message(let msg):
-      emitVerboseTextForm(visitor: &visitor, message: msg, typeURL: _typeURL)
-
-    case .contentJSON(let contentJSON, let options):
-      // If we can decode it, we can write the readable verbose form:
-      if let messageType = Google_Protobuf_Any.messageType(forTypeURL: _typeURL) {
-        do {
-          let m = try unpack(contentJSON: contentJSON,
-                             options: options,
-                             as: messageType)
-          emitVerboseTextForm(visitor: &visitor, message: m, typeURL: _typeURL)
-          return
-        } catch {
-          // Fall through to just print the raw JSON data
-        }
+/// Custom handling for Text format.
+extension AnyMessageStorage {
+      func decodeTextFormat(
+            typeURL url: String, decoder: inout TextFormatDecoder
+      ) throws {
+            // Decoding the verbose form requires knowing the type.
+            _typeURL = url
+            guard
+                  let messageType = Google_Protobuf_Any.messageType(
```

Comment:
This isn't quite what we want. It looks like I allow a break after the `guard` statement, but the other control flow keywords (`if`, etc.) use a space to keep the first clause on the same line. That would fix the weird continuation vs. block clash here.